Common/include/geometry/primal_grid/CPrimalGrid.hpp

```diff
@@ -45,50 +46,60 @@
  */
 class CPrimalGrid {
 protected:
-  unsigned long *Nodes;         /*!< \brief Vector to store the global nodes of an element. */
+  vector<unsigned long> Nodes;         /*!< \brief Global node indices of the element. */
   unsigned long GlobalIndex;    /*!< \brief The global index of an element. */
-  long *Neighbor_Elements;      /*!< \brief Vector to store the elements surronding an element. */
-  short *PeriodIndexNeighbors;  /*!< \brief Vector to store the periodic index of a neighbor.
+  vector<long> Neighbor_Elements;      /*!< \brief Vector to store the elements surronding an element. */
+  vector<short> PeriodIndexNeighbors;  /*!< \brief Vector to store the periodic index of a neighbor.
```

Comment:
This is not optimum with respect to size because std::vector stores 3 pointers to track capacity, however here we do not need "push_back" and so we are wasting 3 pointers.
Also, the "size" of Neighbor_Elements and PeriodIndexNeighbors is the same.

The most efficient way to store PeriodIndexNeighbors is as a static array of N_FACES_MAXIMUM since that will only take 12 bytes (then move it down in the member list). Possibly this could use even less space @vdweide ? (this doc string doesn't add anything to the name of the var...)

For Nodes and Neighbor_Elements use a unique_ptr.